client/process.go

```diff
@@ -84,8 +84,15 @@ func (config *Config) RunProcess(commandLine string, stdin io.Reader, stdout io.
 		}
 
 		// Don't need stdin now we've sent everything. This signals GCS that we are finished sending data.
-		if err := process.Process.CloseStdin(); err != nil {
-			return nil, err
+		err := process.Process.CloseStdin()
+		if err != nil {
+			if !hcsshim.IsNotExist(err) && !hcsshim.IsAlreadyClosed(err) {
+				// This error will occur if the compute system is currently shutting down
+				if perr, ok := err.(*hcsshim.ProcessError); ok && perr.Err != hcsshim.ErrVmcomputeOperationInvalidState {
+					return nil, err
+				}
+			}
+			logrus.Debugf("opengcs: RunProcess %s ignoring legitimate error %v on CloseStdin()", commandLine, err)
```

Comment:
nit: Do we care about logging this? I think it would just generate extra noise.

Reply:
I only put it in to verify my check was trapping. Will remove.